drivers/serial/Kconfig.nrfx

```diff
@@ -30,6 +30,7 @@ config UART_NRFX_UARTE_LEGACY_SHIM
 	depends on UART_NRFX_UARTE
 	# New shim takes more ROM. Until it is fixed use legacy shim in memory
 	# constraint case.
+	default n if SOC_PLATFORM_NRF54L
```

Comment:
This does not make much sense. The default setting is `n` anyway. Instead, we should add `depends on `SOC_PLATFORM_NRF54L`, as the legacy shim cannot be used on nRF54L.

Reply:
Removed